GPT/gpt.py

```diff
@@ -138,27 +138,22 @@ def gpt_apply_prompt(prompt: str, source: str = "", destination: str = ""):
 
         text_to_process: GPTMessageItem = actions.user.gpt_get_source_text(source)
 
-        # If after creating the message there wasn't anything in the text_to_process, set it to None so
-        # we don't send meaningless data to the model
-        if text_to_process.get("text", "") == "":
-            text_to_process = None  # type: ignore
```

Comment:
Unless I am missing other change, we still need to set this here since otherwise it will pass empty messages when using `model please`. Updated the check to check for both images and text and only then setting it to be None.

```suggestion
       if text_to_process.get("text", "") == "" and text_to_process.get("image_url") == "":
            text_to_process = None  # type: ignore
```

Reply:
There is one issue with that - when using a thread you won't always have text you are trying to modify.  You could be asking followup questions about the context that is already in the thread, in which case there would not be any text to modify.  I regularly add things to the context or ask questions with no target text.  Sometimes I do it just to ask a question about general knowledge (i.e. who is Kevin Bacon?)

Reply:
Maybe model please should be a special case that requires the text to be present.

Reply:
Sorry I am not following your comment. `text_to_process` is just a variable that is passed through and eventually gets concatenated to the prompt if it is not `None`. If the user didn't pass in any text or an image we don't want to concatenate an empty message and we can ignore this variable. I don't think this affects threading at all. 

We still can use the thread or context if this variable is None.

Reply:
I couldn't apply your change for some reason, but I did it in the next commit for all prompts other than 'ask'.

Reply:
oh I see what you mean-my bad. I'll change it back